newton/tests/test_viewer_loading_splash.py

```
@@ -0,0 +1,479 @@
+        )
+
+    def test_headless_skips_splash_and_pumps(self):
+        viewer = self._run_init_with_stub(self._args(headless=True))
+        self.assertEqual(viewer.calls, [])
+
+    def test_non_gl_viewer_skips_splash_and_pumps(self):
+        # The ``visible_gl`` gate is ``args.viewer == "gl" and not headless``.
+        # A regression dropping the viewer check would silently pump
+        # frames through a USD/Rerun/Null/Viser viewer.
+        for viewer_kind, viewer_attr, extra in [
+            ("usd", "ViewerUSD", {"output_path": "/tmp/out.usd"}),
```

Ruff (0.15.12)

[error] Probable insecure usage of temporary file or directory: "/tmp/out.usd"

(S108)